### ISSUE TITLE
Include docker autocompletion maybe?

### ISSUE THREAD

**Issue by DoTheEvo** [zimfw/zimfw#345]
it gotten pretty popular.
I can get it working by [following official instruction for non oh-my-zsh](https://docs.docker.com/compose/completion/#zsh), but I kinda like zimfw cuz theres so little additional work needed... I just get it and its done.

**Comment by ericbn** [comment]
Ideally, this should be fixed in the docker installation itself. For example, it's not standard for Docker for macOS. See https://github.com/docker/for-mac/issues/1948

Which package are you installing, @DoTheEvo?

**Comment by DoTheEvo** [comment]
You mean that just by installing docker zsh would have automaticly autocompletion?
Seems strange.
Also not sure what packages you mean, package of what?
I am on arch linux, docker is from official repo, zim is from AUR.

**Comment by ericbn** [comment]
At least the installation should give you a directory with a file named `_docker`, do you can add that to your `fpath`. Do you get that with the official docker for arch linux?

I don't think it's strange that installations would also install the completion functions for the shells, the same way they make the binaries available to be called from the shells too. With homebrew, you usually get the completions installed with the respective formulae. For example, I got the Zsh completions installed for aws (awscli), exa, fd, hub, kubectl (kubernetes-cli), rg (ripgrep), and youtube-dl when I installed them with homebrew.

**Comment by DoTheEvo** [comment]
Nope, you have to do that manually [per their instructions](https://docs.docker.com/compose/completion/#zsh)
But I could imagine that if developers would do it routinely, shit might get slow rahter fast.

And if its not one nice single place where its easy to check out, but anywhere as long as its in fpath... it might get pretty messy pretty fast.

**Comment by ericbn** [comment]
Yes, Homebrew symlinks all completions in one standard directory, so the fpath is still kept short.

**Comment by naibaf0** [comment]
If you check the package contents for the archlinux docker packages you can see that the package in fact does include the autocompletion for zsh: https://www.archlinux.org/packages/community/x86_64/docker/

Works without issues for me and nothing else needs to be done for it to work (except for generally having the autocompletion module enabled).

I do not think that I have ever had to install a zsh plugin to get specific autocompletion on arch, although in the case of gcloud, direnv and some more I had to source the completion files manually  in my zshrc

**Comment by Spriithy** [comment]
Hi, what about debian autocompletion ? I'm kind of a noob so I don't really know

**Comment by Spriithy** [comment]
> Hi, what about debian autocompletion ? I'm kind of a noob so I don't really know

Installing `bash-completion` from apt fixed the autocompletion issue.
